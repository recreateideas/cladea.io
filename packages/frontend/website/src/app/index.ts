export * from "./App";
export * from "./features";
