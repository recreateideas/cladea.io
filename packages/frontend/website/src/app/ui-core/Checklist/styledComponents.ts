import styled from 'styled-components';

export const Container = styled.div`
    width: 100%;
    ul {
        background-color: inherit;
        .MuiListItemIcon-root {
            min-width: auto;
        }
        .MuiButtonBase-root {
        }
    }
`;
