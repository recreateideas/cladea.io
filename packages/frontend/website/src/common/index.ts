export * from './utils';
export * from './hooks';
export * from './converters';
