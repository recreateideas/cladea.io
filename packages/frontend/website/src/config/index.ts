export const apiGatewaUrl = process.env.REACT_APP_API_GATEWAY_URL;

console.log(process.env);
