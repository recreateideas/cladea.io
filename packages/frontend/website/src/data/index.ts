import land110m from "./land-110m.json";

export { land110m };
