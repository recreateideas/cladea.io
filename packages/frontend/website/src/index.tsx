import "./app/ui-core/theme/GlobalStyle/roboto.css";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";
import { store } from "./redux";
import { ServicedApp, ThemeProviderConnected } from "./app";

ReactDOM.render(
  <Provider store={store}>
    <ThemeProviderConnected>
      <ServicedApp />
    </ThemeProviderConnected>
  </Provider>,
  document.getElementById("root")
);
