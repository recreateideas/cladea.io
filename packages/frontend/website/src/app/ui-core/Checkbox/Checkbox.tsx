import { forwardRef, ReactElement, Ref } from 'react';
import { FormControlLabel, Checkbox as CheckboxCore, CheckboxProps } from '@mui/material';
import { Container } from './styledComponents';

interface IProps extends CheckboxProps {
    label?: string | ReactElement;
}
export const Checkbox = forwardRef((props: IProps, ref: Ref<any>): ReactElement => {
    const { label, checked, value } = props;
    const isChecked = !!checked || !!value;
    const Input = <CheckboxCore ref={ref} checked={isChecked} {...props} />;
    return (
        <Container>{label ? <FormControlLabel label={label} control={Input} /> : Input}</Container>
    );
});

Checkbox.displayName = 'Checkbox';

Checkbox.defaultProps = {};
