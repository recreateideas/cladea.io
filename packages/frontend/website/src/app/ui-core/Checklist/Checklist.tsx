import { ReactElement, useState, useEffect, forwardRef, Ref } from 'react';
import { List, ListItem, ListItemButton, ListItemIcon, ListItemText } from '@mui/material';
import { isEqual } from 'lodash';
import { Checkbox } from '../Checkbox';
import { Container } from './styledComponents';

type Value = string | number;
export type CheckedItems = Array<Value>;
type Item = {
    label: string | number | ReactElement;
    value: Value;
    checked?: boolean;
};
interface IProps {
    items: Array<Item>;
    secondaryAction?: ReactElement;
    onChange?: (checkedItems: CheckedItems) => any;
    controlled?: boolean;
    value?: CheckedItems;
    disableRipple?: boolean;
}
export const Checklist = forwardRef((props: IProps, ref: Ref<any>): ReactElement => {
    const {
        controlled = false,
        disableRipple = true,
        value: checkedProp,
        secondaryAction,
        items,
        onChange,
    } = props;
    const [checked, setChecked] = useState<CheckedItems>([]);

    const handleToggle = (value: Value) => () => {
        const currentIndex = checked.indexOf(value);
        const newChecked = [...checked];
        if (currentIndex === -1) {
            newChecked.push(value);
        } else {
            newChecked.splice(currentIndex, 1);
        }
        setChecked(newChecked);
        /* istanbul ignore else */
        if (onChange) {
            onChange(newChecked);
        }
    };
    useEffect(() => {
        if (controlled && checkedProp && !isEqual(checked, checkedProp)) {
            setChecked(checkedProp);
        }
    }, [controlled, checked, checkedProp]);
    return (
        <Container>
            <List sx={{ width: '100%', maxWidth: 360, bgcolor: 'background.paper' }}>
                {items.map((item, i) => {
                    const { label, value } = item;
                    const labelId = `checkbox-list-label-${value}`;
                    return (
                        <ListItem key={i} secondaryAction={secondaryAction} disablePadding>
                            <ListItemButton onClick={handleToggle(value)} dense>
                                <ListItemIcon>
                                    <Checkbox
                                        edge="start"
                                        checked={checked.includes(value)}
                                        disableRipple={disableRipple}
                                        inputProps={{ 'aria-labelledby': labelId }}
                                    />
                                </ListItemIcon>
                                <ListItemText id={labelId} primary={label} />
                            </ListItemButton>
                        </ListItem>
                    );
                })}
            </List>
        </Container>
    );
});

Checklist.displayName = 'Checklist';

Checklist.defaultProps = {};
