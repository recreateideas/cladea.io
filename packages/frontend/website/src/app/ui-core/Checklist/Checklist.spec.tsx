import { Checklist } from './Checklist';
import { Checkbox } from '../Checkbox';
import { ListItemButton } from '@mui/material';
import { TestProvider } from 'src/tests';

const { shallow, mount } = TestProvider();
describe('<Checklist/>', () => {
    const items = [
        {
            value: 1,
            label: <div className="some-test-item">'some label'</div>,
        },
        {
            value: 2,
            label: <div className="some-test-item">'some other label'</div>,
        },
    ];
    it('should render', () => {
        const root = shallow(<Checklist items={items} />);
        expect(root).toMatchSnapshot();
    });
    it('should render the correct number of checkboxes', () => {
        const root = shallow(<Checklist items={items} />);
        const checkboxs = root.find(Checkbox);
        expect(checkboxs.length).toBe(2);
    });
    it("should check the checkbox when the prop is passed and it's controlled", () => {
        const root = mount(<Checklist items={items} value={undefined} controlled={true} />);
        root.setProps({ value: [1] });
        root.update();
        const checkbox = root.find(Checkbox).at(0);
        expect(checkbox.props().checked).toBe(true);
    });
    it('should call the onChange when clicked method if supplied, adding the toggled value', () => {
        const changeSpy = jest.fn();
        const root = mount(<Checklist items={items} controlled={true} onChange={changeSpy} />);
        const checkboxButton = root.find(ListItemButton).at(0);
        checkboxButton.simulate('click');
        root.update();
        expect(changeSpy).toHaveBeenCalledWith([1]);
    });
    it('should call the onChange when clicked method if supplied, removing the toggled value', () => {
        const changeSpy = jest.fn();
        const root = mount(
            <Checklist items={items} value={[1]} controlled={true} onChange={changeSpy} />
        );
        const checkboxButton = root.find(ListItemButton).at(0);
        checkboxButton.simulate('click');
        root.update();
        expect(changeSpy).toHaveBeenCalledWith([]);
    });
    it('should call the onChange when clicked method if suppliedddd', () => {
        const changeSpy = jest.fn();
        const root = mount(<Checklist items={items} controlled={true} onChange={changeSpy} />);
        const checkboxButton = root.find(ListItemButton).at(0);
        checkboxButton.simulate('click');
        root.update();
        const checkbox = root.find(Checkbox).at(0);
        expect(checkbox.props().checked).toBe(true);
    });
});
