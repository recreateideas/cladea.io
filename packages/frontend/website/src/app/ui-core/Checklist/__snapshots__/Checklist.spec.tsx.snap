// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checklist/> should render 1`] = `
<styled.div>
  <ForwardRef(List)
    sx={
      Object {
        "bgcolor": "background.paper",
        "maxWidth": 360,
        "width": "100%",
      }
    }
  >
    <ForwardRef(ListItem)
      disablePadding={true}
      key="0"
    >
      <ForwardRef(ListItemButton)
        dense={true}
        onClick={[Function]}
      >
        <ForwardRef(ListItemIcon)>
          <Checkbox
            checked={false}
            disableRipple={true}
            edge="start"
            inputProps={
              Object {
                "aria-labelledby": "checkbox-list-label-1",
              }
            }
          />
        </ForwardRef(ListItemIcon)>
        <ForwardRef(ListItemText)
          id="checkbox-list-label-1"
          primary={
            <div
              className="some-test-item"
            >
              'some label'
            </div>
          }
        />
      </ForwardRef(ListItemButton)>
    </ForwardRef(ListItem)>
    <ForwardRef(ListItem)
      disablePadding={true}
      key="1"
    >
      <ForwardRef(ListItemButton)
        dense={true}
        onClick={[Function]}
      >
        <ForwardRef(ListItemIcon)>
          <Checkbox
            checked={false}
            disableRipple={true}
            edge="start"
            inputProps={
              Object {
                "aria-labelledby": "checkbox-list-label-2",
              }
            }
          />
        </ForwardRef(ListItemIcon)>
        <ForwardRef(ListItemText)
          id="checkbox-list-label-2"
          primary={
            <div
              className="some-test-item"
            >
              'some other label'
            </div>
          }
        />
      </ForwardRef(ListItemButton)>
    </ForwardRef(ListItem)>
  </ForwardRef(List)>
</styled.div>
`;
